File: src/interface/merge-files-request.ts
export default class MergeFilesRequest {
  rootPath: string
  noSelectedPaths: Array<string>
  excludeDirs: Array<string>
  excludeExts: Array<string>
  enableGitignore: boolean
  
  constructor(rootPath: string, noSelectedPaths: Array<string>, excludeDirs: Array<string>, excludeExts: Array<string>, enableGitignore: boolean) {
    this.rootPath = rootPath
    this.noSelectedPaths = noSelectedPaths
    this.excludeDirs = excludeDirs
    this.excludeExts = excludeExts
    this.enableGitignore = enableGitignore
  }
}

File: src-tauri/src/request/merge_files_request.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MergeFilesRequest {
    pub root_path: String,
    pub no_selected_paths: Vec<String>,
    pub exclude_dirs: Vec<String>,
    pub exclude_exts: Vec<String>,
    pub enable_gitignore: bool,
}
